1.32.1: FTBFS on [armel]

~~~~
[ 37%] Building CXX object src/CMakeFiles/zm.dir/zm_local_camera.cpp.o
cd /<<PKGBUILDDIR>>/dbuild/src && /usr/bin/c++  -DHAVE_SENDFILE4_SUPPORT=1 -I/<<PKGBUILDDIR>>/dbuild/src -I/<<PKGBUILDDIR>>/src -I/<<PKGBUILDDIR>>/dbuild -I/usr/include/arm-linux-gnueabi -I/usr/include/mysql  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -std=c++11 -Wall -D__STDC_CONSTANT_MACROS -O2   -std=gnu++11 -o CMakeFiles/zm.dir/zm_local_camera.cpp.o -c /<<PKGBUILDDIR>>/src/zm_local_camera.cpp
/tmp/ccLN2vHS.s: Assembler messages:
/tmp/ccLN2vHS.s:527: Error: selected processor does not support `vdup.8 q12,r12' in ARM mode
/tmp/ccLN2vHS.s:529: Error: selected processor does not support `vldm r5!,{q0,q1,q2,q3}' in ARM mode
/tmp/ccLN2vHS.s:530: Error: selected processor does not support `vldm r6!,{q4,q5,q6,q7}' in ARM mode
/tmp/ccLN2vHS.s:533: Error: selected processor does not support `vrshl.u8 q8,q0,q12' in ARM mode
/tmp/ccLN2vHS.s:534: Error: selected processor does not support `vrshl.u8 q9,q1,q12' in ARM mode
/tmp/ccLN2vHS.s:535: Error: selected processor does not support `vrshl.u8 q10,q2,q12' in ARM mode
/tmp/ccLN2vHS.s:536: Error: selected processor does not support `vrshl.u8 q11,q3,q12' in ARM mode
/tmp/ccLN2vHS.s:537: Error: selected processor does not support `vrshl.u8 q4,q4,q12' in ARM mode
/tmp/ccLN2vHS.s:538: Error: selected processor does not support `vrshl.u8 q5,q5,q12' in ARM mode
/tmp/ccLN2vHS.s:539: Error: selected processor does not support `vrshl.u8 q6,q6,q12' in ARM mode
/tmp/ccLN2vHS.s:540: Error: selected processor does not support `vrshl.u8 q7,q7,q12' in ARM mode
/tmp/ccLN2vHS.s:541: Error: selected processor does not support `vsub.i8 q4,q4,q8' in ARM mode
/tmp/ccLN2vHS.s:542: Error: selected processor does not support `vsub.i8 q5,q5,q9' in ARM mode
/tmp/ccLN2vHS.s:543: Error: selected processor does not support `vsub.i8 q6,q6,q10' in ARM mode
/tmp/ccLN2vHS.s:544: Error: selected processor does not support `vsub.i8 q7,q7,q11' in ARM mode
/tmp/ccLN2vHS.s:545: Error: selected processor does not support `vadd.i8 q4,q4,q0' in ARM mode
/tmp/ccLN2vHS.s:546: Error: selected processor does not support `vadd.i8 q5,q5,q1' in ARM mode
/tmp/ccLN2vHS.s:547: Error: selected processor does not support `vadd.i8 q6,q6,q2' in ARM mode
/tmp/ccLN2vHS.s:548: Error: selected processor does not support `vadd.i8 q7,q7,q3' in ARM mode
/tmp/ccLN2vHS.s:549: Error: selected processor does not support `vstm r7!,{q4,q5,q6,q7}' in ARM mode
/tmp/ccLN2vHS.s:2591: Error: selected processor does not support `vldm r0!,{q0,q1,q2,q3}' in ARM mode
/tmp/ccLN2vHS.s:2592: Error: selected processor does not support `vldm r1!,{q4,q5,q6,q7}' in ARM mode
/tmp/ccLN2vHS.s:2595: Error: selected processor does not support `vabd.u8 q0,q0,q4' in ARM mode
/tmp/ccLN2vHS.s:2596: Error: selected processor does not support `vabd.u8 q1,q1,q5' in ARM mode
/tmp/ccLN2vHS.s:2597: Error: selected processor does not support `vabd.u8 q2,q2,q6' in ARM mode
/tmp/ccLN2vHS.s:2598: Error: selected processor does not support `vabd.u8 q3,q3,q7' in ARM mode
/tmp/ccLN2vHS.s:2599: Error: selected processor does not support `vstm r2!,{q0,q1,q2,q3}' in ARM mode
/tmp/ccLN2vHS.s:5211: Error: selected processor does not support `vdup.32 q8,r12' in ARM mode
/tmp/ccLN2vHS.s:5213: Error: selected processor does not support `vldm r0!,{q0,q1,q2,q3}' in ARM mode
/tmp/ccLN2vHS.s:5214: Error: selected processor does not support `vldm r1!,{q4,q5,q6,q7}' in ARM mode
/tmp/ccLN2vHS.s:5217: Error: selected processor does not support `vabd.u8 q0,q0,q4' in ARM mode
/tmp/ccLN2vHS.s:5218: Error: selected processor does not support `vabd.u8 q1,q1,q5' in ARM mode
/tmp/ccLN2vHS.s:5219: Error: selected processor does not support `vabd.u8 q2,q2,q6' in ARM mode
/tmp/ccLN2vHS.s:5220: Error: selected processor does not support `vabd.u8 q3,q3,q7' in ARM mode
/tmp/ccLN2vHS.s:5221: Error: selected processor does not support `vrshr.u8 q0,q0,#3' in ARM mode
/tmp/ccLN2vHS.s:5222: Error: selected processor does not support `vrshr.u8 q1,q1,#3' in ARM mode
/tmp/ccLN2vHS.s:5223: Error: selected processor does not support `vrshr.u8 q2,q2,#3' in ARM mode
/tmp/ccLN2vHS.s:5224: Error: selected processor does not support `vrshr.u8 q3,q3,#3' in ARM mode
/tmp/ccLN2vHS.s:5225: Error: selected processor does not support `vmul.i8 q0,q0,q8' in ARM mode
/tmp/ccLN2vHS.s:5226: Error: selected processor does not support `vmul.i8 q1,q1,q8' in ARM mode
/tmp/ccLN2vHS.s:5227: Error: selected processor does not support `vmul.i8 q2,q2,q8' in ARM mode
/tmp/ccLN2vHS.s:5228: Error: selected processor does not support `vmul.i8 q3,q3,q8' in ARM mode
/tmp/ccLN2vHS.s:5229: Error: selected processor does not support `vpadd.i8 d0,d0,d1' in ARM mode
/tmp/ccLN2vHS.s:5230: Error: selected processor does not support `vpadd.i8 d2,d2,d3' in ARM mode
/tmp/ccLN2vHS.s:5231: Error: selected processor does not support `vpadd.i8 d4,d4,d5' in ARM mode
/tmp/ccLN2vHS.s:5232: Error: selected processor does not support `vpadd.i8 d6,d6,d7' in ARM mode
/tmp/ccLN2vHS.s:5233: Error: selected processor does not support `vpadd.i8 d0,d0,d0' in ARM mode
/tmp/ccLN2vHS.s:5234: Error: selected processor does not support `vpadd.i8 d1,d2,d2' in ARM mode
/tmp/ccLN2vHS.s:5235: Error: selected processor does not support `vpadd.i8 d2,d4,d4' in ARM mode
/tmp/ccLN2vHS.s:5236: Error: selected processor does not support `vpadd.i8 d3,d6,d6' in ARM mode
/tmp/ccLN2vHS.s:5237: Error: selected processor does not support `vst4.32 {d0[0],d1[0],d2[0],d3[0]},[r2]!' in ARM mode
Manifying 22 pod documents
make[3]: *** [src/CMakeFiles/zm.dir/build.make:287: src/CMakeFiles/zm.dir/zm_image.cpp.o] Error 1
~~~~

https://buildd.debian.org/status/fetch.php?pkg=zoneminder&arch=armel&ver=1.32.1-1&stamp=1539059554&raw=0

--- a/src/zm_image.cpp
+++ b/src/zm_image.cpp
@@ -3317,11 +3317,11 @@ __attribute__((noinline)) void std_fastblend(const uint8_t* col1, const uint8_t*
 }
 
 /* FastBlend Neon for AArch32 */
-#if (defined(__arm__) && !defined(ZM_STRIP_NEON))
+#if (defined(__arm__) && !defined(__armel__) && !defined(ZM_STRIP_NEON))
 __attribute__((noinline,__target__("fpu=neon")))
 #endif
 void neon32_armv7_fastblend(const uint8_t* col1, const uint8_t* col2, uint8_t* result, unsigned long count, double blendpercent) {
-#if (defined(__arm__) && !defined(ZM_STRIP_NEON))
+#if (defined(__arm__) && !defined(__armel__) && !defined(ZM_STRIP_NEON))
   static int8_t divider = 0;
   static double current_blendpercent = 0.0;
 
@@ -3708,11 +3708,11 @@ __attribute__((noinline)) void std_delta8_abgr(const uint8_t* col1, const uint8_
 }
 
 /* Grayscale Neon for AArch32 */
-#if (defined(__arm__) && !defined(ZM_STRIP_NEON))
+#if (defined(__arm__) && !defined(__armel__) && !defined(ZM_STRIP_NEON))
 __attribute__((noinline,__target__("fpu=neon")))
 #endif
 void neon32_armv7_delta8_gray8(const uint8_t* col1, const uint8_t* col2, uint8_t* result, unsigned long count) {
-#if (defined(__arm__) && !defined(ZM_STRIP_NEON))
+#if (defined(__arm__) && !defined(__armel__) && !defined(ZM_STRIP_NEON))
 
   /* Q0(D0,D1)   = col1+0 */
   /* Q1(D2,D3)   = col1+16 */
@@ -3784,11 +3784,11 @@ __attribute__((noinline)) void neon64_armv8_delta8_gray8(const uint8_t* col1, co
 }
 
 /* RGB32 Neon for AArch32 */
-#if (defined(__arm__) && !defined(ZM_STRIP_NEON))
+#if (defined(__arm__) && !defined(__armel__) && !defined(ZM_STRIP_NEON))
 __attribute__((noinline,__target__("fpu=neon")))
 #endif
 void neon32_armv7_delta8_rgb32(const uint8_t* col1, const uint8_t* col2, uint8_t* result, unsigned long count, uint32_t multiplier) {
-#if (defined(__arm__) && !defined(ZM_STRIP_NEON))
+#if (defined(__arm__) && !defined(__armel__) && !defined(ZM_STRIP_NEON))
 
   /* Q0(D0,D1)   = col1+0 */
   /* Q1(D2,D3)   = col1+16 */

